wgpu-core/src/device/resource.rs

```diff
@@ -2889,7 +2890,8 @@ impl<A: HalApi> Device<A> {
                 }
 
                 if let Some(ref interface) = shader_module.interface {
-                    shader_expects_dual_source_blending = interface.is_fragment_entry_dual_source(fragment);
+                    shader_expects_dual_source_blending =
+                        interface.is_fragment_entry_dual_source(fragment).expect("Internal error: Fragment entrypoint should not be set in function if not present in shader interface");
```

Comment:
It's fine by me if you don't want this expect (or want to reword it), but currently it has to be here as the function it calls can fail if the entry point for some reason should not exist.